### ISSUE TITLE
The result of using the predict_classes() is much lower than the accuracy of using the evaluate()

### ISSUE THREAD

**Issue by BeCuriousCat** [fchollet/keras#5999]
Hi,everybody.I use keras in my project which is to predict the punctuation for chinese. 
Firstly,i get some classed chinese corpus.
Secondly,i use a Python segment tool to cut the corpus.
Thirdly,i use a modified one hot encode function to  replace chinese words with int number.
Then i use these date to train my lstm neural networks.
But finally  when i try to check the result ,i find the evaluate() gave more high accuracy than predict_classes().
Can anyone help?
